python_digest/__init__.py

```diff
@@ -109,8 +117,11 @@ def calculate_nonce(timestamp, secret, salt=None):
     '''
     if not salt:
         salt = ''.join([random.choice('0123456789ABCDEF') for x in range(4)])
-    return b"%s:%s:%s" % (timestamp, salt,
-                         md5.md5(("%s:%s:%s" % (timestamp, salt, secret)).encode('utf-8')).hexdigest())
+    return "%s:%s:%s" % (
```

Comment:
Same as above, changing the return type of this method seems like it could break downstream expectations.

Reply:
```calculate_nonce``` is used in ```build_digest_challenge``` and ```validate_nonce```, both of which require text to work correctly.  I discovered this when running tests.